### PR TITLE
docs: (RPC) add info about matching hono versions client-server

### PR DESCRIPTION
**Pull request by catalinpit**
Add information about using Hono RPC in projects where the client and server are in a single project. I spent way too much time wrestling with this before finding a comment online about the Hono version.

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @catalinpit !

Thank you for the PR. This will be helpful.

I think we must add the description, not be in the `Compile your code before using it (recommended)` section. You can create a new section about it at the same level as `Compile your code before using it (recommended)`:

CleanShot 2024-12-21 at 16 48 25@2x

Because the problem is not only when the code is being compiled; other problems will happen without compiling. For example, suppose you have monorepo, which includes one server application and one frontend. In that case, if the versions are different and, you use the exported type from the server app in the frontend, the editor throws the error like this:

CleanShot 2024-12-21 at 16 54 14@2x

Repro: https://github.com/yusukebe/hono-version-mismatch

This problem occurs not only due to a version mismatch but also when the paths to the `hono` files referenced by each project are different. However, in many cases, ensuring that the versions are the same can solve the problem.

**Comment by catalinpit** [comment]
Hey @yusukebe, thanks for the reply and suggestion! I actually ran into that problem as well - `Type instantiation is excessively deep and possibly infinite`.

I'll add a new section!

**Comment by catalinpit** [comment]
Done. Change made. 🙌

File: docs/guides/rpc.md [hunk 1]
```diff
@@ -523,8 +533,6 @@ const res = await client.posts.$post({
 
 If your project is a monorepo, this solution does fit well. Using a tool like [`turborepo`](https://turbo.build/repo/docs), you can easily separate the server project and the client project and get better integration managing dependencies between them. Here is [a working example](https://github.com/m-shaka/hono-rpc-perf-tips-example).
 
-If your client and server are in a single project, [project references](https://www.typescriptlang.org/docs/handbook/project-references.html) of `tsc` is a good option.
```
Comment (yusukebe) [inline docs/guides/rpc.md:526]: Should we remove this line? I think it's OK to leave it.

**Comment by yusukebe** [comment]
@catalinpit 

Sorry for the late response. I've added the comment!

File: docs/guides/rpc.md (hunk 1 above)
Comment (catalinpit) [reply docs/guides/rpc.md:526]: I added it above in its section `#### TypeScript project references`. Should I leave it here as well?

**Comment by catalinpit** [comment]
> @catalinpit
> 
> Sorry for the late response. I've added the comment!

No worries. Thanks for the replies!

File: docs/guides/rpc.md (hunk 1 above)
Comment (yusukebe) [reply docs/guides/rpc.md:526]: Ahhhh, I missed it! Let's go with this.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@catalinpit 

Looks good. Merging now. Thank you!

**Comment by catalinpit** [comment]
> @catalinpit
> 
> Looks good. Merging now. Thank you!

Thanks a bunch!

**Pull request merged**